lib/display.rb

```diff
@@ -28,8 +35,33 @@ def choose_cell
     @console.present(@message.prompt_for_cell)
   end
 
-  def receive_integer
-    @console.receive.to_i
+  def get_valid_cell
+    choice = receive_integer
+    until @validator.valid_cell?(choice)
+      choose_cell
+      choice = receive_integer
+    end
+    choice
+  end
+
+  def valid_game_mode_response
+    response = receive_integer
+    valid_options = [1, 2, 3]
+    until valid_options.include?(response)
+      choose_game_mode
+      response = receive_integer
+    end
+    response
+  end
+
+  def valid_play_again_response
+    response = receive_lower_case_character.downcase
```

Comment:
`receive_lower_case_character.downcase` indicates the character may not be lower case ;)